src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -918,12 +886,12 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
             // When returning, the `Next` method will set `next` to the state to goto on the subsequent invocation.
             // This is similar to `Task.ContinueWith`, but since we have a fixed number of states we can avoid
             // the overhead of actually using `Task.ContinueWith`.
-            var next = State.InvokeBegin;
+            var next = State.ResourceInnerBegin;
 
             // The `scope` tells the `Next` method who the caller is, and what kind of state to initialize to
             // communicate a result. The outermost scope is `Scope.Invoker` and doesn't require any type
```

Comment:
You can feel free to prune all of this and just refer to the comment in the base type.